src/Datastore/EntityMapper.php

```diff
@@ -40,6 +40,23 @@ class EntityMapper
     private $encode;
 
     /**
+     * @var array
+     */
+    private $valueTypes = [
+        "nullValue",
+        "booleanValue",
+        "integerValue",
+        "doubleValue",
+        "timestampValue",
+        "keyValue",
+        "stringValue",
+        "blobValue",
+        "geoPointValue",
+        "entityValue",
+        "arrayValue",
+    ];
```

Comment:
Not without reading the discovery document. Do you have any suggestions on a way to reliably determine the value type without looking at a predetermined list? I could use `strpos()` I guess.